bot/exts/evergreen/cheatsheet.py

```diff
@@ -0,0 +1,93 @@
+import random
+import re
+import typing as t
+from urllib.parse import quote_plus
+
+from discord import Embed
+from discord.ext import commands
+from discord.ext.commands import BucketType, Context
+
+from bot.constants import Channels, Colours, ERROR_REPLIES
+
+ERROR_MESSAGE = """
+Unknown cheat sheet. Please try to reformulate your query.
+
+**Examples**:
+```md
+.cht read json
+.cht hello world
+.cht lambda
+```
+If the problem persists send a message in <#{channel}>
+"""
+
+
+class ChtSh(commands.Cog):
+    """Commands that sends a result of a cht.sh search in code blocks."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @staticmethod
+    def fmt_error_embed() -> Embed:
+        """If the cht.sh search returned 404, overwrite it to send a custom error embed."""
+        embed = Embed(colour=Colours.soft_red)
+        embed.title = random.choice(ERROR_REPLIES)
+        embed.description = ERROR_MESSAGE.format(channel=Channels.dev_contrib)
+        return embed
+
+    def result_fmt(self, url: str, body_text: str) -> t.Tuple[bool, t.Union[str, Embed]]:
+        """Format Result."""
+        if body_text.startswith("#  404 NOT FOUND"):
+            embed = self.fmt_error_embed()
+            return True, embed
+
+        body_space = min(1986 - len(url), 1000)
+
+        if len(body_text) > body_space:
+            description = f"**Result Of cht.sh**\n" \
```

Comment:
Could do: Make successfully found cheat sheets also output in embeds. This would remove the need for the isEmbed var, and may make the output look better?

Reply:
> Could do: Make successfully found cheat sheets also output in embeds. This would remove the need for the isEmbed var, and may make the output look better?

Embeds have a limit on number of characters on a single line hence the code block will break.  This is why i didn't add it in a embed.

Reply:
Ahh yes, we have a line paginator which would be useful here. But it hasn't been ported over from the other bot yet. See https://github.com/python-discord/sir-lancebot/pull/555